src/Type/TypeCombinator.php

```diff
@@ -200,8 +201,7 @@ public static function union(Type ...$types): Type
 			if ($types[$i] instanceof StringType && !$types[$i] instanceof ClassStringType) {
 				$hasGenericScalarTypes[ConstantStringType::class] = true;
 			}
-			$enumCases = $types[$i]->getEnumCases();
```

Comment:
If the issue is with `getEnumCases` be slow.

Shouldn't be introduced a new method `isSomething`, implemented by all types in order to avoid the error message
```
Doing instanceof PHPStan\\\\Type\\\\Enum\\\\EnumCaseObjectType is error\\-prone and deprecated\\
``` 
?

Then
```
if ($types[$i]->isSomething()) {
```

Reply:
Not sure its worth a new type-method just for this bottleneck